fcrepo-api-x-loader/src/main/java/org/fcrepo/apix/loader/impl/LoaderRoutes.java

```diff
@@ -90,7 +90,7 @@ public void setRouting(final RoutingFactory routing) {
     @Override
     public void configure() throws Exception {
 
-        from("jetty:http://0.0.0.0:{{loader.port}}/load" +
+        from("jetty:http://{{loader.host}}:{{loader.port}}/load" +
```

Comment:
The one thing about this is that Jetty can get unnecessarily picky about accepting connections that come from network interfaces that don't match.  So if `loader.host` resolves to 1.2.3.4, it won't accept requests from within the container (127.0.0.1).  `0.0.0.0` means accept from everywhere.  Let me try deploy in Docker to check this...

Reply:
So I checked this in Docker, and it works OK in it, with caveats:
* If you use 0.0.0.0 as a host (the default), the loader extension registers itself as `http://0.0.0.0:8081/load`.  This is the URI of the loader service as gets recorded in the API-X registry.  Somehow, it actually works out;  the jetty http client must assume `0.0.0.0` means localhost`
* If API-X is running on a machine with one network interface, any correct value of `loader.host` works fine.
* If API-X is running on a machine with more than one interface, then the jetty component will only bind to the interface that corresponds to `loader.host`.  Or at least that'd the way it's documented to behave.  I wasn't able to test that.  

I think the best solution is to use two variables:  `loader.listen` and `loader.host`, where `loader.listen` defaults to `0.0.0.0`.  We can either do that in this PR, or a new one.  Thoughts @christopher-johnson?

Reply:
`0.0.0.0` is typically considered a "wildcard" meaning: "bind to all available network interfaces", which includes `localhost`

Reply:
@esm Right, but it's being used in two contexts.  Binding (for which `0.0.0.0` makes sense), and in URLs.  (in which it does not make sense).  So literally in this _second_ context, it's saying "my URL is http://0.0.0.0:8081/load`; use that to connect to me".

Reply:
@birkland yup!

Reply:
@birkland The question I have about using 0.0.0.0 is does this limit the connections to IP4?  If you use a hostname, DNS can provide a quad A record if available.  I do not know what can be gained by wildcarding a very specific listener endpoint.  

docker networking configs can vary, but I suspect that a normal approach would be to include services on the same bridge network so that they can talk to the loader by container name:port.  I would not consider exposing the loader port to any external or public interface.   The self-loader that uses "localhost" will only work if 0.0.0.0 remains, that is true.  However, if it knows who it is, it can get its options the same way other services register themselves.  I also think that for sanity, it makes sense to make the loader configuration a model for other services.

BTW, is it worth considering changing camel-jetty to [ camel-jetty9](https://github.com/apache/camel/blob/master/components/camel-jetty9/src/main/java/org/apache/camel/component/jetty9/JettyHttpEndpoint9.java)?  AFAIK, Jetty 9 should be used with Karaf 4.x,, and it looks like the Jetty9 binding does a few things differently than the [default binding](https://github.com/apache/camel/blob/master/components/camel-jetty-common/src/main/java/org/apache/camel/component/jetty/DefaultJettyHttpBinding.java)
At any rate, I do not think that camel or jetty care much about the socket layer as this is delegated to JSR-356 ...

Reply:
Using netstat,  I compare in this file
[netstat APIX IPv6: IPv4 wildcard analysis.txt](https://github.com/fcrepo4-labs/fcrepo-api-x/files/1270810/netstat.APIX.IPv6.IPv4.wildcard.analysis.txt)
 a listener bound to `0.0.0.0` in an IPv6 docker network to one that was bound to a hostname.  Both work... 

I see that routing also has a `{{listener.host}}` set to `0.0.0.0`.  If the intent is to have a single listener configuration for multiple services, then I agree with you that it should be the same variable name.  But if each service is contracted to provide its own listener, then the host should be uniquely configurable for that service.  

In my routing configuration, I map the blueprint configuration file routing listen host variable:
`apix.listen.host = ${apixListenHost}`
  to be the same as the loader host environment variable.. 
 `apixListenHost=${env:APIX_LOADER_HOST:-apix}` in the global properties file.

Reply:
I'm not sure I follow this conversation, but @birkland is proposing two separate properties:
* `loader.listen` defaults to `0.0.0.0` (listen on all interfaces), but can be changed if desired
* `loader.host` defaults to `localhost` and is used to carry the IP address of the loader extension

That seems to provide maximum configurability and reasonable defaults.  Is there a reason @christopher-johnson why those properties wouldn't be sufficient for your use case?

Reply:
@emetsger  The main change here is to remove `localhost` and replace it with the `loader.host` property.  AFAICT, replacing `0.0.0.0` with  `loader.listen` (or `loader.host`) is inconsequential unless it is misconfigured by the user.  I had a question about the use of `0.0.0.0` as a default `service.listen` value working with IPv6, but I guess that I have answered that.  Would you recommend that every `service` follow the same pattern of `service.listen` and `service.host`?  And, how exactly is `service.listen` different from `service.host`?  This seems confusing to me.